source/slang/slang-check-decl.cpp

```diff
@@ -9562,8 +9562,31 @@ namespace Slang
         void visitTargetSwitchStmt(TargetSwitchStmt* stmt)
         {
             CapabilitySet set;
-            for (auto targetCase : stmt->targetCases)
-            {
+            auto targetCaseCount = stmt->targetCases.getCount();
+            for (Index targetCaseIndex = 0; targetCaseIndex < targetCaseCount; targetCaseIndex++)
+            {
+                // We may revieve a `default:` case for a `__target_switch`. If this is the case,
+                // we must resolve the target capability as `any_target` to hint to the capability 
+                // system that the following function must be defined on all codegen targets. If
+                // we do not handle `default:`, the codegen will fail when trying to find a specific
+                // codegen target not handled explicitly by a `case` statment.
+                
+                // We must also ensure the `default` case is last so we have priority to hit `case` statments
+                // without complex logic. We must shift all case statments when doing this logic instead of 
+                // swapping with the final element since some capabilities have overlapping targets.
+                // As a result order matters when resolving `case` statments
+                if (CapabilityName(stmt->targetCases[targetCaseIndex]->capability) == CapabilityName::__default_case)
```

Comment:
Can we avoid introducing this __default_case as a capability?
We should keep capability definitions clean and only about actual targets and capabilities.

It seems the goal here is just to tell if a case is a default case. You can simply check if `capability` is `0` or something like that.

Reply:
That was the original implementation, I will revert to the original implementation.